Content.Server/Anomaly/Effects/SecretDataAnomalySystem.cs

```diff
@@ -0,0 +1,43 @@
+using Content.Server.Anomaly.Components;
+using Robust.Shared.Random;
+
+namespace Content.Server.Anomaly.Effects;
+
+public sealed class SecretDataAnomalySystem : EntitySystem
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<SecretDataAnomalyComponent, MapInitEvent>(OnMapInit);
+    }
+
+    private void OnMapInit(EntityUid uid, SecretDataAnomalyComponent anomaly, MapInitEvent args)
+    {
+        RandomizeSecret(uid,_random.Next(anomaly.RandomStartSecretMin, anomaly.RandomStartSecretMax), anomaly);
+    }
+
+    public void RandomizeSecret(EntityUid uid, int count, SecretDataAnomalyComponent? component = null)
+    {
+        if (!Resolve(uid, ref component))
+            return;
+
+        component.Secret.Clear();
+
+        var variants = new List<AnomalySecretData> {
+            AnomalySecretData.Severity,
+            AnomalySecretData.Stability,
+            AnomalySecretData.OutputPoint,
+            AnomalySecretData.ParticleDanger,
+            AnomalySecretData.ParticleUnstable,
+            AnomalySecretData.ParticleContainment,
+            AnomalySecretData.ParticleTransformation,
+            AnomalySecretData.Behavior,
+        };
```

Comment:
use Enum.GetValues<T>()

Reply:
![image](https://github.com/space-wizards/space-station-14/assets/96445749/df85b364-3e79-4936-904c-c0d49c968675)
its not work